packages/runtime/container-runtime/src/dataStoreContexts.ts

```diff
@@ -6,20 +6,24 @@
 import { IDisposable, ITelemetryBaseLogger, ITelemetryLogger } from "@fluidframework/common-definitions";
 import { assert, Deferred, Lazy } from "@fluidframework/common-utils";
 import { ChildLogger } from "@fluidframework/telemetry-utils";
-import { FluidDataStoreContext } from "./dataStoreContext";
+import { FluidDataStoreContext, LocalFluidDataStoreContext } from "./dataStoreContext";
 
  export class DataStoreContexts implements Iterable<[string,FluidDataStoreContext]>, IDisposable {
-    public readonly notBoundContexts = new Set<string>();
+    private readonly notBoundContexts = new Set<string>();
 
-    // Attached and loaded context proxies
+    /** Attached and loaded context proxies */
     private readonly _contexts = new Map<string, FluidDataStoreContext>();
-    // List of pending contexts (for the case where a client knows a store will exist and is waiting
-    // on its creation). This is a superset of contexts.
-    private readonly contextsDeferred = new Map<string, Deferred<FluidDataStoreContext>>();
+
+    /**
+     * List of pending context needing to be bound, for the case where a client
+     * knows a store will exist and is waiting on its creation.
+     * This also includes bound contexts, in the completed state, so this is a superset of contexts.
+     */
+    private readonly deferredContextBinds = new Map<string, Deferred<FluidDataStoreContext>>();
```

Comment:
context's can also be deferred when waiting for a remote context to show up

Reply:
Thanks for clarifying.